src/scrabble.js

```diff
@@ -1,6 +1,55 @@
+const POINTS = {
+  A: 1, E: 1, I: 1, O: 1, U: 1, L: 1, N: 1, R: 1, S: 1, T: 1,
+  D: 2, G: 2, B: 3, C: 3, M: 3, P: 3, F: 4, H: 4, V: 4, W: 4, Y: 4, K: 5,
+  J: 8, X: 8, Q: 10, Z: 10, '}': 0, '{': 0, '[': 0, ']': 0
+};
+
 class Scrabble {
-  // Write your implementation here
+  constructor(string) {
+    this.string = string
+  }
+  score() {
+    let sum = 0
+
+    if (this.string === null) {
+      return 0
+    }
+
+    let word = this.string.toUpperCase()
 
+    for (let n = 0; n < word.length; n++) {
+      if (/[A-Z]/.test(word[n])) {
+        sum += POINTS[word[n]]
+      }
+      else if (word[0] === '{' && word[word.length - 1] === '}') {
+        return double(word)
+      }
+      else if (word[0] === '[' && word[word.length - 1] === ']') {
+        return triple(word)
+      }
+      else if (word[n] == '{') {
+        sum += POINTS[word[n + 1]]
+      }
+      else if (word[n] == '[') {
+        sum += POINTS[word[n + 1]] * 2
+      }
+    }
+    return sum;
+  }
+}
+function double(word) {
```

Comment:
I thought of it tho indeed it would have looked more encapsulated (which is better) but in that case, I need to call it a separated method and I was aiming to handle all cases in one call.